src/server/game/Entities/Object/Object.cpp

```diff
@@ -2234,9 +2234,21 @@ void WorldObject::_GetNearPoint(float &x, float &y, float &z, float distance, fl
     UpdateAllowedPositionZ(x, y, z);
 }
 
-void WorldObject::GetNearPoint(float &x, float &y, float &z, float distance, float absAngle) const
+void WorldObject::GetNearPoint(float &x, float &y, float &z, float distance, float absAngle, bool distanceIn3D /*= false*/) const
 {
     _GetNearPoint(x, y, z, distance, absAngle);
+    const float ACCEPTABLE_ERROR = 0.5f;
```

Comment:
Instead of doing something like this you could calculate a proper `distance` calculation that considers height difference from `this` object and the object that `requests` the GetNearPoint.